pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -882,16 +888,33 @@ def scatterClicked(self, plt, points, ev):
     def scatterHovered(self, plt, points, ev):
         self.sigPointsHovered.emit(self, points, ev)
 
-    def viewRangeChanged(self):
-        # view range has changed; re-plot if needed
-        self._viewRangeWasChanged = True
-        if( self.opts['clipToView']
-            or self.opts['autoDownsample']
-        ):
-            self.xDisp = self.yDisp = None
-            self.updateItems(styleUpdate=False)
-        elif self.opts['dynamicRangeLimit'] is not None:
-            # update, but do not discard cached display data
+    # def viewTransformChanged(self):
+    #   """ view transform (and thus range) has changed, replot if needed """
+    # viewTransformChanged is only called when the cached viewRect of GraphicsItem
+    # has already been invalidated. However, responding here will make PlotDataItem
+    # update curve and scatter later than intended.
+    #   super().viewTransformChanged() # this invalidates the viewRect() cache!
+        
+    def viewRangeChanged(self, vb=None, ranges=None, changed=None):
+        """ view range has changed; re-plot if needed """
+        update_needed = False
+        if changed is None or changed[0]: 
```

Comment:
changes to horizontal axis require a recalculation if auto-downsampling or clipping are active.